great_tables/_locations.py

```diff
@@ -823,6 +824,41 @@ def resolve_rows_i(
     )
 
 
+def resolve_mask_i(
+    data: GTData | list[str],
+    expr: PlExpr,
+    excl_stub: bool = True,
+    excl_group: bool = True,
+) -> list[tuple[int, int, str]]:
+    """Return data for creating `CellPos`, based on expr"""
+    if not isinstance(expr, PlExpr):
+        raise ValueError("Only Polars expressions can be passed to the `mask` argument.")
+
+    stub_var = data._boxhead.vars_from_type(ColInfoTypeEnum.stub)
+    group_var = data._boxhead.vars_from_type(ColInfoTypeEnum.row_group)
+    cols_excl = [*(stub_var if excl_stub else []), *(group_var if excl_group else [])]
+
+    frame: PlDataFrame = data._tbl_data
+    df = frame.select(expr)
```

Comment:
What if the rules for the mask were as follows?:

* the mask expressions should--when run by `frame.select(mask_expr)`--return a DataFrame that:
  - columns: has the same or fewer columns (names must match original frame, but could be in a different order).
  - rows: has the same number of rows
* after running the expression, we validate this right away
* the mask is assumed to be in the original order
* in the case we can capture the column name and row number of each cell (e.g. with `enumerate(mask_result.iter_rows())`).

Reply:
Thanks for summarizing the rules.  

It seems that for the last part, we only need a single loop to gather all the information required by `CellPos`.

Reply:
How about this version?  

For prototyping, use `assert` for validation, which can later be replaced with `raise ValueError()`.

Reply:
I've replaced `assert` with `raise ValueError()` and added more test cases.